bindings/wasm/src/account/traits.rs

```diff
@@ -0,0 +1,223 @@
+  #[wasm_bindgen(method)]
+  pub fn key_insert(
+    this: &WasmStorage,
+    did: WasmDID,
+    location: WasmKeyLocation,
+    private_key: String,
+  ) -> PromisePublicKey;
+  #[wasm_bindgen(method)]
+  pub fn key_get(this: &WasmStorage, did: WasmDID, location: WasmKeyLocation) -> PromisePublicKey;
+  #[wasm_bindgen(method)]
+  pub fn key_del(this: &WasmStorage, did: WasmDID, location: WasmKeyLocation) -> PromiseUnit;
+  #[wasm_bindgen(method)]
+  pub fn key_sign(this: &WasmStorage, did: WasmDID, location: WasmKeyLocation, data: Vec<u8>) -> PromiseSignature;
+  #[wasm_bindgen(method)]
+  pub fn key_exists(this: &WasmStorage, did: WasmDID, location: WasmKeyLocation) -> PromiseBool;
+  #[wasm_bindgen(method)]
+  pub fn published_generation(this: &WasmStorage, did: WasmDID) -> PromiseOptionGeneration;
+  #[wasm_bindgen(method)]
+  pub fn set_published_generation(this: &WasmStorage, did: WasmDID, index: WasmGeneration) -> PromiseUnit;
+  #[wasm_bindgen(method)]
+  pub fn chain_state(this: &WasmStorage, did: WasmDID) -> PromiseOptionChainState;
+  #[wasm_bindgen(method)]
+  pub fn set_chain_state(this: &WasmStorage, did: WasmDID, chain_state: WasmChainState) -> PromiseUnit;
+  #[wasm_bindgen(method)]
+  pub fn state(this: &WasmStorage, did: WasmDID) -> PromiseOptionIdentityState;
+  #[wasm_bindgen(method)]
+  pub fn set_state(this: &WasmStorage, did: WasmDID, state: WasmIdentityState) -> PromiseUnit;
+  #[wasm_bindgen(method)]
+  pub fn purge(this: &WasmStorage, did: WasmDID) -> PromiseUnit;
+}
+
+impl Debug for WasmStorage {
+  fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+    f.write_str("WasmStorage")
+  }
+}
+
+#[async_trait::async_trait(?Send)]
+impl Storage for WasmStorage {
+  /// Sets the account password.
+  async fn set_password(&self, password: EncryptionKey) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.set_password(password.into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Write any unsaved changes to disk.
+  async fn flush_changes(&self) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.flush_changes());
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Attempt to obtain the exclusive permission to modify the given `did`.
+  /// The caller is expected to make no more modifications after the lease has been dropped.
+  /// Returns an [`IdentityInUse`][crate::Error::IdentityInUse] error if already leased.
+  async fn lease_did(&self, did: &IotaDID) -> AccountResult<DIDLease> {
+    let promise: Promise = Promise::resolve(&self.lease_did(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    // workaround due to problems deserializing `DIDLease`, even with serde "rc" feature
+    let account_result: AccountResult<bool> = result.into();
+    account_result.map(|value| DIDLease::from(AtomicBool::from(value)))
+  }
```

Comment:
It's not necessary to serialize the lease because it doesn't need to be stored, it just needs to exist in memory. Using just a bool won't work correctly, because while in JS you can set `"did:iota:123...": true` to store that it's in use, it's never going to be set back to `false` when the lease is dropped in Rust. Something like that should be all that's needed:

```rust
#[wasm_bindgen(js_name = DIDLease)]
pub struct WasmDIDLease(pub(crate) DIDLease);

#[wasm_bindgen]
impl WasmDIDLease {
  #[wasm_bindgen(constructor)]
  pub fn new() -> Self {
    WasmDIDLease(DIDLease::new())
  }

  #[wasm_bindgen(constructor)]
  pub fn load(&self) -> bool {
    self.0.load()
  }

  #[wasm_bindgen(constructor)]
  pub fn store(&self, value: bool) {
    self.0.store(value)
  }
}
```

That way, when the lease is dropped, the inner `AtomicBool` is set to false which should enable the correct behavior. But we need to make sure the `WasmDIDLease` isn't dropped to early somewhere. I'm probably missing some other subtleties, but with tests or an example we should see if it works.